source/Octopus.Manager.Tentacle/Proxy/ProxyWizardModel.cs

```diff
@@ -153,48 +153,51 @@ public IEnumerable<CommandLineInvocation> GenerateScript()
-                    .Argument("proxyUsername", ProxyUsername)
-                    .Argument("proxyPassword", ProxyPassword)
-                    .Argument("proxyHost", string.Empty)
-                    .Argument("proxyPort", string.Empty)
-                    .Build();
-            }
-            else if (ProxyConfigType == ProxyConfigType.CustomProxy)
-            {
-                var host = new UriBuilder(ProxyServerHost).Host;
-
-                yield return Cli(Command)
-                    .Argument("proxyEnable", true)
-                    .Argument("proxyUsername", ProxyUsername)
-                    .Argument("proxyPassword", ProxyPassword)
-                    .Argument("proxyHost", host)
-                    .Argument("proxyPort", ProxyServerPort)
-                    .Build();
+                case ProxyConfigType.NoProxy:
+                    yield return Cli(Command)
+                        .Argument("proxyEnable", false)
+                        .Argument("proxyUsername", string.Empty)
+                        .Argument("proxyPassword", string.Empty)
+                        .Argument("proxyHost", string.Empty)
+                        .Argument("proxyPort", string.Empty)
+                        .Build();
+                    break;
+                case ProxyConfigType.DefaultProxy:
+                    // Explicitly clear the credentials
+                    yield return Cli(Command)
+                        .Argument("proxyEnable", true)
+                        .Argument("proxyUsername", string.Empty)
+                        .Argument("proxyPassword", string.Empty)
+                        .Argument("proxyHost", string.Empty)
+                        .Argument("proxyPort", string.Empty)
+                        .Build();
+                    break;
+                case ProxyConfigType.DefaultProxyCustomCredentials:
+                    yield return Cli(Command)
+                        .Argument("proxyEnable", true)
+                        .Argument("proxyUsername", ProxyUsername)
+                        .Argument("proxyPassword", ProxyPassword)
+                        .Argument("proxyHost", string.Empty)
+                        .Argument("proxyPort", string.Empty)
+                        .Build();
+                    break;
+                case ProxyConfigType.CustomProxy:
+                {
+                    var host = new UriBuilder(ProxyServerHost).Host;
+
+                    yield return Cli(Command)
+                        .Argument("proxyEnable", true)
+                        .Argument("proxyUsername", ProxyUsername)
+                        .Argument("proxyPassword", ProxyPassword)
+                        .Argument("proxyHost", host)
+                        .Argument("proxyPort", ProxyServerPort)
+                        .Build();
+                    break;
+                }
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
This is a change of behaviour.

Previously, if the `ProxyConfigType` was an unsupported value, the enumeration would not return a value at this point. The new beavhiour will throw an exception in the same situation.

This change in behaviour would not be an issue right now, as all values of the enum are supported. However, if a new value is added to the enum, it could throw here instead of silently skipping over it.

Reply:
I think whoever adds a new `ProxyConfigType` needs to be aware that they need to implement changes to this method too. So this change of behaviour would ensure they do.